cli/internal/run/run.go

```diff
@@ -77,128 +78,149 @@ func (rs *runSpec) ArgsForTask(task string) []string {
-                         mechanism to identify which packages have changed.
-  --team                 The slug or team ID of the remote cache team.
-  --token                A bearer token for remote caching. You can also set
-                         the value of the current token by setting an
-                         environment variable named TURBO_TOKEN.
-  --ignore               Files to ignore when calculating changed files
-                         (i.e. --since). Supports globs.
-  --profile              File to write turbo's performance profile output into.
-                         You can load the file up in chrome://tracing to see
-                         which parts of your build were slow.
-  --parallel             Execute all tasks in parallel. (default false)
-  --include-dependencies Include the dependencies of tasks in execution.
-                         (default false)
-  --no-deps              Exclude dependent task consumers from execution.
-                         (default false)
-  --no-cache             Avoid saving task results to the cache. Useful for
-                         development/watch tasks. (default false)
-  --output-logs          Set type of process output logging. Use full to show
-                         all output. Use hash-only to show only turbo-computed
-                         task hashes. Use new-only to show only new output with
-                         only hashes for cached tasks. Use none to hide process
-                         output. (default full)
-  --dry/--dry-run[=json] List the packages in scope and the tasks that would be run,
-                         but don't actually run them. Passing --dry=json or
-                         --dry-run=json will render the output in JSON format.
-  --remote-only		     Ignore the local filesystem cache for all tasks. Only
-                         allow reading and caching artifacts using the remote cache.
-`)
-	return strings.TrimSpace(helpText)
+	cmd := getCmd(c.Config, c.Ui, c.Processes)
+	return util.HelpForCobraCmd(cmd)
 }
 
 // Run executes tasks in the monorepo
 func (c *RunCommand) Run(args []string) int {
-	startAt := time.Now()
-	log.SetFlags(0)
-	flags := flag.NewFlagSet("run", flag.ContinueOnError)
-	flags.Usage = func() { c.Config.Logger.Info(c.Help()) }
-	if err := flags.Parse(args); err != nil {
-		return 1
-	}
-
-	opts, err := parseRunArgs(args, c.Config, c.Ui)
+	cmd := getCmd(c.Config, c.Ui, c.Processes)
+	cmd.SetArgs(args)
+	err := cmd.Execute()
 	if err != nil {
+		exitErr := &process.ChildExit{}
+		if errors.As(err, &exitErr) {
+			return exitErr.ExitCode
+		}
 		c.logError(c.Config.Logger, "", err)
 		return 1
 	}
+	return 0
+}
+
+type run struct {
+	opts      *Opts
+	config    *config.Config
+	ui        cli.Ui
+	processes *process.Manager
+}
 
-	ctx, err := context.New(context.WithGraph(c.Config, opts.cacheOpts.Dir))
+func (r *run) run(targets []string) error {
+	startAt := time.Now()
```

Comment:
This feels slightly more-focused and maps better to tracking the time of the command. A good change.